PlayRho/Dynamics/World.cpp

```diff
@@ -2111,6 +2111,7 @@ ContactCounter World::FindNewContacts()
     // to eliminate any node pairs that have the same body here before the key pairs are
     // sorted.
     for_each(cbegin(m_proxies), cend(m_proxies), [&](ProxyId pid) {
+        if (pid == DynamicTree::GetInvalidSize()) return;
```

Comment:
Thank you super much for recognizing this problem! Looks like you've made two different PRs which can address the fundamental problem you'd discovered: `m_proxies` can have invalid IDs in it before `FindNewContacts` is called.

I'm interested in unit test code that checks for this problem and am working on at least one check now for this. Sorry this wasn't caught already by a unit test. Glad at least it was caught by an assert.

Reply:
Thanks again for your help with this! I believe this is fixed now with pull request #341 that I've merged into master. If you can confirm, I'd very much appreciate it!